spec/factories/host.rb

```diff
@@ -6,30 +6,37 @@
     ipaddress           { "127.0.0.1" }
     user_assigned_os    { "linux_generic" }
     power_state         { "on" }
-  end
 
-  factory :host_with_ref, :parent => :host do
-    sequence(:ems_ref) { |n| "host-#{seq_padded_for_sorting(n)}" }
+    transient do
+      authtype { nil }
+      storage_count { nil }
+    end
+
+    trait :with_ref do
+      ems_ref_type { "HostSystem" }
+      sequence(:ems_ref) { |n| "host-#{seq_padded_for_sorting(n)}" }
```

Comment:
I know factories should only set what is necessary, but kinda feels like these should always be set.

Reply:
> I know factories should only set what is necessary, but kinda feels like ems_ref should always be set.

I wonder if this was because of host impi provisioning where some hosts were present that weren't from EmsRefresh.

If we no longer support that then I agree with moving ems_ref into the base factory (doesn't have to be decided right away could be future cleanup)

Reply:
thnx for telling me I am not crazy

Reply:
Well I wouldn't go that far 😉